Test for establishing a connection with SUT after a connection reset

shivansh opened the pull request:
Testing for behavior of TCP stack when injecting a packet with **RST** bit set.

tuexen committed:
Merge pull request #72 from hoelscher/notification_stopped_event

Notification stopped event

tuexen committed:
Update SCTP_SocketAPI_Status.md

tuexen committed:
Fix Linux compilation.

hoelscher committed:
add support for sctp_adaptation_indication_parameter in init chunk

hoelscher committed:
styling changes

hoelscher committed:
updates from review

hoelscher committed:
add support for cmsghdr for sendmsg syscall

hoelscher committed:
add testcase for sendmsg

hoelscher committed:
add check funktion for cmsghdr struct

hoelscher committed:
implement cmsg for sctp_sndrcv and sctp_init

hoelscher committed:
add testcases

hoelscher committed:
fix error for SCTP?INIT

hoelscher committed:
change ifdef

hoelscher committed:
modify testcase

hoelscher committed:
merge from master

hoelscher committed:
modify testcase

hoelscher committed:
add support for cmsg sctp_nxtinfo and sctp_rcvinfo

hoelscher committed:
merge with nplab

hoelscher committed:
add syntax for sctp_extrcvinfo

hoelscher committed:
add support for sctp_extrcvinfo

hoelscher committed:
remove prints

hoelscher committed:
change testcase recvmsg

tuexen committed:
Merge pull request #74 from hoelscher/cmsgs_for_sendmsg

Cmsgs for sendmsg

tuexen committed:
Get it compiling on NetBSD.

tuexen committed:
Merge pull request #75 from hoelscher/cmsg_sctp_init

Cmsg sctp init

tuexen committed:
Fix compilation on Linux.

tuexen committed:
Merge pull request #76 from hoelscher/recvmsg_cmsg

Recvmsg cmsg

tuexen committed:
Fix compilation on Linux.

While there remove trainling whitespaces.

tuexen committed:
Allow various flags to be expressions.

tuexen committed:
Add symbols for serinfo_next_flags.

tuexen committed:
Improve tests.

tuexen committed:
Remove trailing whitespaces.

tuexen committed:
Use fixed names.

tuexen committed:
Cleanups. No functional change. Still don't know why it works...

tuexen committed:
Allow ... as parameters in sndrcvinfo when receiving.

tuexen committed:
Fix size_t issues.

TheAomx committed:
injecting custom hb-information with heartbeat-ack-chunk now possible

TheAomx committed:
adjustments in order to use already existing flag.

TheAomx committed:
removed unwanted whitspace

tuexen committed:
Merge pull request #79 from TheAomx/master

injecting custom hb-information with heartbeat-ack-chunk now possible

tuexen committed:
Get things compiling on FreeBSD (64 bit).

tuexen committed:
Improving compilation.

tuexen committed:
More fixes...

hoelscher committed:
add support for sctp_bindx with only [...] as addresses

hoelscher committed:
add support for sctp_connectx

hoelscher committed:
add sctp_peeloff

hoelscher committed:
Merge branch 'sctp_bindx' of github.com:hoelscher/packetdrill into sctp_bindx

hoelscher committed:
add sctp_send syscall

hoelscher committed:
add sctp_sendx syscall

Commit:
switch syntax for sctp_connectx

tuexen committed:
Merge pull request #80 from hoelscher/sctp_bindx

sctp_bindx

tuexen committed:
Merge pull request #81 from hoelscher/sctp_send

Sctp send

tuexen committed:
Merge pull request #82 from hoelscher/sctp_sendx

Sctp sendx

tuexen committed:
Get it compiling on Linux.

tuexen committed:
Fix compilation on NetBSD.

tuexen committed:
Update SCTP_SocketAPI_Status.md

tuexen committed:
Update SCTP_SocketAPI_Status.md

hoelscher committed:
add support for sctp_freepaddrs and sctp_getpaddrs

hoelscher committed:
add support for sctp_freeladdrs and sctp_getladdrs

hoelscher committed:
fix cast of size_t

hoelscher committed:
add srto_assoc_id

hoelscher committed:
Merge remote-tracking branch 'origin/master' into review_sctp_datastructures

hoelscher committed:
add sse_assoc_id in shutdown_event

tuexen committed:
Merge pull request #83 from hoelscher/last_syscalls

Last syscalls

TheAomx committed:
fix for issue 77

TheAomx committed:
Merge branch 'master' of https://github.com/nplab/packetdrill

hoelscher committed:
add se_assoc_id for SCTP_EVENT [gs]etsockopt

TheAomx committed:
adjustments

hoelscher committed:
add sasoc_assoc_id in struct sctp_assocparams

TheAomx committed:
fixed one small typo

TheAomx committed:
simplified if condition

tuexen committed:
Merge pull request #84 from TheAomx/master

Fix for Issue 77

tuexen committed:
Whitespace change.

hoelscher committed:
add [gs]etsockopt SCTP_PRIMARY_ADDR

hoelscher committed:
add spp_assoc_id for sctp_paddrparams

hoelscher committed:
add assoc_id in MAXSEG socketoption

hoelscher committed:
add sack_assoc_id in for structure sctp_sack_info_expr

hoelscher committed:
add sctp_default_prinfo for [gs]etsockopt syscall

hoelscher committed:
add sstat_assoc_id in struct sctp_status

hoelscher committed:
add spinfo_assoc_id in struct sctp_paddrinfo

hoelscher committed:
change checks for assoc ids in run_systemcall and modify tests

hoelscher committed:
modify testcase

hoelscher committed:
add first testcases for testing  packetdrill for sctp_ap_api

hoelscher committed:
add first api testcases for socketoptions

hoelscher committed:
add notifications to api_tests

hoelscher committed:
add test for cmsgs in api_test

hoelscher committed:
add test for cmsgs with recvmsg

hoelscher committed:
add last test for unimplemented socketoptions

hoelscher committed:
add testcases for sctp_sendv

hoelscher committed:
add syntax with and without sctp_assoc_id for all sockopt structures

hoelscher committed:
modify tests for tests without assoc_id in cmsgs

hoelscher committed:
modify tests for tests without assoc_id for notifications

hoelscher committed:
add some more test for api test script

hoelscher committed:
remove check for sinfo_assoc_id for sendcalls

hoelscher committed:
add support for socketoption sctp_auth_active_key

hoelscher committed:
add support for socketoption sctp_autoclose

hoelscher committed:
add support for socketoption sctp_disable_fragments

hoelscher committed:
add support for socketoption sctp_default_send_param

hoelscher committed:
add symbol for sctp_i_want_mapped_v4_addr for syscall [gs]etsockopt

TheAomx committed:
first commit regarding support of generic packets

hoelscher committed:
add support for sctp_hmac_ident sockopt

TheAomx committed:
fixed shifting to correct value

hoelscher committed:
add symbol for sctp_reuse_port, but test doesn't work

hoelscher committed:
add symbol for sctp_explicit_eor

hoelscher committed:
add symbol for socketopt SCTP_PARTIAL_DELIVERY_POINT,

hoelscher committed:
add symbol SCTP_AUTO_ASCONF for socketoption

hoelscher committed:
add symbol SCTP_CONTEXT for socketoption

hoelscher committed:
add symbol SCTP_GET_ASSOC_NUMBER for socketoption

hoelscher committed:
add support for sctp_get_assoc_id_lists for [gs]etsockopt, shoult end in an error but didn't do taht for one-to-one socket

hoelscher committed:
add last symbols for sockopts on freeBSD

hoelscher committed:
add support for sctp_peer/local_auth_chunks in getsockopt, tested only with empty chunklist

hoelscher committed:
add defines for SCTP_AUTH_DELETE_KEY and SCTP_AUTH_DEACTIVATE_KEY

hoelscher committed:
add support for socketoption SCTP_SET_PEER_PRIMARY_ADDR, test does not work

hoelscher committed:
add last two structures in script.c/h

hoelscher committed:
add syntax in parser for sctp_authchunk and sctp_authkey

TheAomx committed:
adjusted braces at if

tuexen committed:
Merge pull request #85 from hoelscher/review_sctp_datastructures

Add assoc_ids and cleanup.

tuexen committed:
Fix compilation on 64-bit FreeBSD.

TheAomx committed:
refactored structure to support generic packets instead of generic chunks

TheAomx committed:
fix for alignment

TheAomx committed:
removed unused definition

tuexen committed:
Merge pull request #87 from hoelscher/apitests

Apitests

yweweler committed:
Fixed example to work with the latest folder structure.

tuexen committed:
Merge pull request #88 from yweweler/master

Fixed example to work with the latest folder structure.

tuexen committed:
Merge pull request #89 from hoelscher/syntax_for_assoc_id

Syntax for assoc

tuexen committed:
Add support for SCTP_EXPLICIT_EOR socket option.

tuexen committed:
Merge branch 'master' of https://github.com/nplab/packetdrill

Commit:
add support for sctp_auth_chunk socketoption

hoelscher committed:
add sockot SCTP_AUTH_KEY

hoelscher committed:
modify testcase for sctp_local_auth_chunks

tuexen committed:
Merge pull request #90 from hoelscher/last_sockopts

Last sockopts

TheAomx committed:
fixes for support of generic init messages.

TheAomx committed:
style fix

TheAomx committed:
adjustments

TheAomx committed:
tab fix at parser.y...

TheAomx committed:
indentation...

TheAomx committed:
some adjustments

TheAomx committed:
first impl of os specific ifdefs in lexer

TheAomx committed:
optimized lexer rules

TheAomx committed:
refactored lexer

TheAomx committed:
added free to early returns...

TheAomx committed:
added block init of vars like required in ansi c

tuexen committed:
Merge pull request #91 from TheAomx/master

Implementation of generic packets

TheAomx committed:
Merge branch 'master' of https://github.com/nplab/packetdrill

tuexen committed:
Ensure that getsocket actually checks the return value.

TheAomx committed:
refactord structure of lexer a bit

TheAomx committed:
modified lexer to match case insestive the os name

hoelscher committed:
add Symbols for linux sockopts

hoelscher committed:
add api_tests for linux

hoelscher committed:
fix testcase for bsd

hoelscher committed:
fix testcase for bsd

hoelscher committed:
fix testcase for bsd

tuexen committed:
Merge pull request #93 from hoelscher/linux_syscalls

Linux syscalls and sockopts

TheAomx committed:
Merge branch 'master' of https://github.com/nplab/packetdrill

tuexen committed:
Merge pull request #92 from TheAomx/master

Implementation of platform specific ifdef lexer rules

TheAomx committed:
possible fix for out of the blue packets

TheAomx committed:
refactored handle_listen_for_script_packet method

TheAomx committed:
small fix

hoelscher committed:
fix len for cmsg on linux

hoelscher committed:
fix testcases for sctp_recvmsg

hoelscher committed:
fix testcase for cmsg sctp_sndrcv for sendmsg

hoelscher committed:
fis testcase for cmsg sctp_sndrcv for recvmsg

TheAomx committed:
fixed support for ootb packets for udp test cases

TheAomx committed:
adjustment for ipv6

tuexen committed:
Merge pull request #94 from TheAomx/oob_packet_support

Out of the blue packet support

tuexen committed:
Merge pull request #95 from hoelscher/linuxapitest

Linuxapitest

hoelscher committed:
fix for bsd for run_packet.c

hoelscher committed:
add testcase for sctp_enable_stream_reset

hoelscher committed:
add symbols for freeBSD for sctp_enable_stream_reset

hoelscher committed:
add symols for sctp_reset_streams

hoelscher committed:
add support for sctp_reset_streams for setsockopt

hoelscher committed:
add one testcase for sctp_reset_streams

hoelscher committed:
add symbol for sctp_reset_assoc for setsockopt

hoelscher committed:
add testcase for sctp_reset_assoc

hoelscher committed:
remove prints

hoelscher committed:
add support for sctp_add_streams

hoelscher committed:
add symbol for sctp_add_streams

hoelscher committed:
add testcase for sctp_add_streams

hoelscher committed:
add support for sctp_stream_reset_event

hoelscher committed:
add symbols for Flags of sctp_stream_reset_event

hoelscher committed:
add fix for handling ipv6 in rum_packet.c

hoelscher committed:
delete SCTP_PEER_ADDR_PARAMS from testcases for sctp_reconfig

hoelscher committed:
add support for sctp_assoc_reset_event

hoelscher committed:
add support for sctp_stream_change_event

hoelscher committed:
add symbols for sctp_stream_change_event

tuexen committed:
Merge pull request #96 from hoelscher/sctp_enable_stream_reset

Add support for stream reset socket options.

hoelscher committed:
add support for reconfig chunk

hoelscher committed:
add testcases for reconfig chunk

hoelscher committed:
Merge branch 'master' into sctp_reconfig_events

hoelscher committed:
fix for sctp_reconfiguration socketoptions and tests

hoelscher committed:
merge part1

tuexen committed:
Whitspace changes.

tuexen committed:
Merge pull request #97 from hoelscher/sctp_reconfig_events

Sctp reconfig events

tuexen committed:
Free only allocated memory.

hoelscher committed:
remove printf

hoelscher committed:
Merge branch 'master' into reconfig_chunks

hoelscher committed:
add support for outgoing ssn reset request parameter

hoelscher committed:
change testcase for injecting outgoing ssn reset request

hoelscher committed:
add RECONFIG_RESPONSE to testcases

hoelscher committed:
change syntax of reconfig parameter

hoelscher committed:
add support for reconfig_response parameter

hoelscher committed:
prepare for optional check Parameter values

hoelscher committed:
add support for ssn_tsn_reset_request parameter for sctp_reconfig chunk

hoelscher committed:
add testcases for ssn_tsn_reset_request

hoelscher committed:
add support for incoming_ssn_reset_request_parameter for reconfig_chunk

hoelscher committed:
add testcases for incoming_ssn_reset_request

hoelscher committed:
add support for sctp_add_incoming_streams_request_parameter for sctp_reconfig

hoelscher committed:
add testcase for add_incoming_streams

hoelscher committed:
add support for add_outgoing_streams parameter

hoelscher committed:
add testcase for add_outgoing_streams

hoelscher committed:
made sctp_reconfig parameter values for outgoing packets optional

hoelscher committed:
delete whitespaces

hoelscher committed:
delete another whitespace

hoelscher committed:
delete one more whitespace

tuexen committed:
Merge pull request #98 from hoelscher/reconfig_chunks

Reconfig chunks

tuexen committed:
Merge pull request #99 from hirenp/master

A simple server-side test that sends exactly an initial window (IW10)…

tuexen committed:
Merge pull request #100 from hirenp/master

Add 1) dupack triggered fast-retransmit 2) partial ack processing scr…

shivansh committed:
Test for establishing a connection after a reset

hirenp commented on gtests/net/packetdrill/tests/bsd/tcp/listen/listen-incoming-ack.pkt:
```diff
@@ -0,0 +1,22 @@
+// Test behavior when a listener gets an incoming packet that has
+// the ACK bit set but not the SYN bit set.
```
Test behavior when a listener gets and incoming packet from new sender on a non-established session that doesn't have a SYN bit set but has an ACK bit set. Listener should send a RESET.

hirenp commented:
Looks good otherwise.

shivansh commented:
Updated the comments, tested and working.

tuexen committed:
Use UNRECOGNIZED_CHUNK_TYPE instead of UNRECOGNIZED_CHUNK.

hoelscher committed:
fix for bug in sids list for OUTGOING-/INCOMING_SSN_RESET_REQUEST

hoelscher committed:
revert change from fix

tuexen committed:
Merge pull request #102 from hoelscher/master

OUTGOING-/INCOMING_SSN_RESET_REQ FIX

hoelscher committed:
fix for srs_flags in struct sctp_reset_streams

hoelscher committed:
fix for one test

hoelscher committed:
fix for reconfig_response parameter fields sender_next_tsn, receiver_next_tsn

hoelscher committed:
fix handling of sctp_stream_reset_event.strreset_stream_list

hoelscher committed:
tsn in struc sctp_reset_assoc_event set to script_values

hoelscher committed:
add empty reconfig chunk

hoelscher committed:
fix empty reconfig chunk

hoelscher committed:
remove prints

hoelscher committed:
remove tailing whitespace

hoelscher committed:
fix for reconfig_response

hoelscher committed:
second fix for ... in sender_next_tsn and receiver_next_tsn

hoelscher committed:
add sctp_reset_stream for getsockopt

hoelscher committed:
allow all sctp_parameter in reconfig_chunk

tuexen committed:
Merge pull request #104 from hoelscher/master

Fix for SCTP_RECONFIGURATIONs

hoelscher committed:
fix reconfig parameter for invalid parameter tests with generic_chunk type

hoelscher committed:
Merge remote-tracking branch 'upstream/master'

hoelscher committed:
fix for invalid reconfig parameters

hoelscher committed:
insert rfc6525 in SCTP_SocketAPI_Status.md

hoelscher committed:
Update SCTP_SocketAPI_Status.md

hoelscher committed:
fix chunk_to_string for reconfig parameter

hoelscher committed:
modify print of INCOMING_SSN_RESET parameter

hoelscher committed:
Merge branch 'master' of https://www.github.com/hoelscher/packetdrill

hoelscher committed:
add reserved fields for ADD_OUTGOING_STREAMS and ADD_INCOMING_STREAMS to print

hoelscher committed:
fix for sctp_reconfig_response

tuexen committed:
Provide names for the threads.

This helps in debugging.

tuexen committed:
Add a function to check if a thread is sleeping.

This is required to support blocking system calls on FreeBSD.
Sponsored by Netflix.

tuexen committed:
Fix compilation on Linux.

Sponsored by Netflix

tuexen committed:
Merge pull request #105 from hoelscher/master

Fixes for sctp stream reconfiguration
